File: CS/Sample/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace Sample.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult GridViewPartial() {
            var model = Enumerable.Range(0, 10).Select(i => new { ID = i, Text = "Text " + i, Checked = i%2 == 0});
            return PartialView(model);
        }
    }
}

File: VB/Sample/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc

Namespace Sample.Controllers
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function GridViewPartial() As ActionResult
			Dim model = Enumerable.Range(0, 10).Select(Function(i) New With {Key .ID = i, Key .Text = "Text " & i, Key .Checked = i Mod 2 = 0})
			Return PartialView(model)
		End Function
	End Class
End Namespace